qiskit/circuit/library/arithmetic/adders/classicaladder.py

```diff
@@ -0,0 +1,119 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals
+
+"""Module-level docstring describing what the file content is."""
+
+from qiskit.circuit import QuantumCircuit, QuantumRegister, AncillaRegister
+
+
+class ClassicalAdder(QuantumCircuit):
+    r"""A ClassicalAdd Circuit to perform in-place addition on two qubit registers.
+     Circuit to compute the sum of two qubit registers using the Classical Addition Part from [1].
+
+     Given two equally sized input registers that store quantum states
+    :math:`|a\rangle` and :math:`|b\rangle`, performs addition of numbers that
+    can be represented by the states, storing the resulting state in-place in the second register:
+    .. math::
+        |a\rangle |b\rangle \mapsto |a\rangle |a+b\rangle
+    Here :math:`|a\rangle` (and correspondingly :math:`|b\rangle`) stands for the direct product
+    :math:`|a_n\rangle \otimes |a_{n-1}\rangle \ldots |a_{1}\rangle \otimes |a_{0}\rangle`
+    which denotes a quantum register prepared with the value :math:`a = 2^{0}a_{0} + 2^{1}a_{1} +
+    \ldots 2^{n}a_{n}` [2].
```

Comment:
What's [2]? Another reference? If yes, that one is missing below in the References.